src/lambdas/RedriveSNSDQLEvents/helper.ts

```diff
@@ -0,0 +1,76 @@
+import { SQSBatchItemFailure, SQSEvent } from 'aws-lambda'
+import { AuditEvent } from '../../types/auditEvent'
+import { tryParseJSON } from '../../utils/helpers/tryParseJson'
+
+export type ProcessingResult = {
+  sqsMessageId: string
+  failed: boolean
+  statusReason: string
+  auditEvent?: AuditEvent
+}
+
+export const SQSBatchItemFailureFromProcessingResultArray = (
+  processingResultArray: ProcessingResult[]
+): SQSBatchItemFailure[] => {
+  return processingResultArray.map((element) => {
+    return { itemIdentifier: element.sqsMessageId }
+  })
+}
+
+export const generateEventIdLogMessageFromProcessingResult = (
+  processingResultArrays: ProcessingResult[][]
```

Comment:
I think here, if it is possible to have a flat input array, would remove one of the nested loops from the function.